### PR TITLE
chore: move ParseRepoPath to internal package

### PR DESCRIPTION
**Pull request by qweeah**
Putting it in `cmd/oras/internal` will trigger code coverage scan on command declaration code, which effects our code coverage.

### CODE REVIEW

**Review by TerryHowe — APPROVED** [review]
LGTM

**Comment by qweeah** [comment]
I am not sure why this doesn't trigger codcov auto comment. The coverage rate increases ~7% in https://app.codecov.io/gh/oras-project/oras/pulls.

File: internal/repository/repository.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+package repository
+
+import (
+	"fmt"
+	"strings"
+
+	"oras.land/oras-go/v2/registry"
+)
+
+func ParseRepoPath(rawReference string) (hostname, namespace string, err error) {
```
Comment (shizhMSFT) [inline internal/repository/repository.go:24]: Missing documentation.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**